asset-api-management/apis/svc-asset-details/extension/clients/asset_details_client.go

```diff
@@ -0,0 +1,48 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"svc-asset-details-layer/models"
+)
+
+type AssetDetailsClient struct {
+	rdsClient *RdsClient
+}
+
+const (
+	docIDKeyword   = "_docid.keyword"
+	docTypeKeyword = "docType.keyword"
+)
+
+func NewAssetDetailsClient(configuration *Configuration) *AssetDetailsClient {
+	return &AssetDetailsClient{rdsClient: NewRdsClient(configuration)}
+}
+
+func (c *AssetDetailsClient) GetAssetDetails(ctx context.Context, ag string, targetType string, assetId string) (*models.AssetDetails, error) {
+	if len(strings.TrimSpace(assetId)) == 0 {
+		return nil, fmt.Errorf("assetId must be present")
+	}
+
+	result, err := FetchAssetDetails(ctx, ag, targetType, assetId, 1)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sourceArr := (*result)["hits"].(map[string]interface{})["hits"].([]interface{})
+	if len(sourceArr) > 0 {
+		assetDetails := sourceArr[0].(map[string]interface{})
+		return &models.AssetDetails{
+			AccountId:       fmt.Sprintf("%v", assetDetails["accountId"]),
+			Source:          assetDetails["source"].(string),
+			SourceName:      assetDetails["sourceName"].(string),
+			TargetType:      assetDetails["targetType"].(string),
+			Tags:            assetDetails["tags"].(map[string]string),
+			PrimaryProvider: fmt.Sprintf("%v", assetDetails["rawData"]),
```

Comment:
@ershad-paladin This implementation assumes that all assets will have the rawData field or, soon, the primary_provider field added by the delta engine. However, there are legacy assets that do not include this field. For those legacy assets, we need to manually construct the primary_provider field by subtracting the known common fields from the asset data. The remaining data is then placed into the primary_provider object.

Reply:
Now handled legacy assets data model also